src/app/create/[id]/createBlueprintSteps/PatternDetails.tsx

```diff
@@ -83,8 +87,17 @@ const PatternDetails = ({
         }
         title="Upload test .eml"
         helpText="Our AI will autofill fields based on contents inside your mail. Don't worry you can edit them later"
-        setFile={(e) => {
-          console.log('setting the file');
+        setFile={async (e) => {
+          if (!e) return;
+
+          try {
+            const response = await findOrCreateDSP(e);
+            console.log(response);
+          } catch (error) {
+            toast.error('Failed to find or create DSP');
```

Comment:
Let's be more user friendly and not use jargon like DSP in user facing toasts. Make this a full english sentence. Also this shouldn't error, it should warn because you can still make regexes without the DKIM signature passing

Reply:
Updated